app/views/shared/_resource_details.html.erb

```diff
@@ -122,9 +122,9 @@
           <%= link_to t(".actions.deregister"), "#" %>
         </li>
         <% end %>
-        <% if resource.is_a?(WasteExemptionsEngine::TransientRegistration) %>
+        <% if resource.is_a?(WasteExemptionsEngine::TransientRegistration) && can?(:update, resource) %>
```

Comment:
What are your thoughts on moving this to a helper e.g. `can_resume?(resource)`. Just conscious we use it in a couple of places and this might

- centralise the logic
- make the intent clearer in the view

Reply:
Makes sense to me – I was wondering about the same thing but got lazy. Will have a look!